handlers/apps.go

```diff
@@ -59,7 +61,22 @@ func AppList(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if criteria.IncludeHealth {
-		rateInterval, err := adjustRateInterval(r.Context(), business, p.Namespace, p.RateInterval, p.QueryTime)
+		// When the cluster is not specified, we need to get it. If there are more than one,
+		// get the one for which the namespace creation time is oldest
+		clusters, _ := business.Namespace.GetNamespaceClusters(r.Context(), p.Namespace)
+		if len(clusters) == 0 {
+			err = fmt.Errorf("No clusters found for namespace  [%s]", p.Namespace)
+			handleErrorResponse(w, err, err.Error())
+			return
+		}
+		var cluster string
+		var creationTimestamp time.Time
+		for i, cl := range clusters {
+			if i == 0 || cl.CreationTimestamp.Before(creationTimestamp) {
```

Comment:
Where does `creationTimestamp` get initialized? If it's uninitialized then I think you'll always get `cluster = clusters[0]` because of the first condition `i == 0`. This would be a good thing to write a test for where there's more than one namespace and the rate interval gets adjusted according to the oldest.

Reply:
Actually it looks like there's a helper func `GetOldestNamespace` that is implemented differently. Perhaps you can just use that here instead?

Reply:
Sure, update to use GetOldestNamespace